libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Selectors/MostSpecificSelector.cs

```diff
@@ -31,14 +31,11 @@ public class MostSpecificSelector : ITriggerSelector
         [JsonProperty("selector")]
         public ITriggerSelector Selector { get; set; }
 
-        [Newtonsoft.Json.JsonIgnore]
-        public IExpressionParser Parser { get; set; } = new ExpressionEngine(TriggerTree.LookupFunction);
-
         public void Initialize(IEnumerable<OnCondition> conditionals, bool evaluate)
         {
             foreach (var conditional in conditionals)
             {
-                _tree.AddTrigger(conditional.GetExpression(Parser), conditional);
+                _tree.AddTrigger(conditional.GetExpression(), conditional);
```

Comment:
>GetExpression [](start = 45, length = 13)

How does the registration of ignore/optional happen?

Reply:
it's in the builtinfunction table

---
In reply to: [376551521](https://github.com/microsoft/botbuilder-dotnet/pull/3350#discussion_r376551521) [](ancestors = 376551521)